JBPM-5336 Add test coverage for container aliases

MarianMacik opened the pull request:
Hi,

here are additional tests which cover alias functionality as described in BPMSPL-543. Tests focus on the following:

Process Service
- starting processes within containers with different GAVs using the same aliases
- signalling processes
- setting/getting process variables
- completing work items

User Task Service
- getting input/output of tasks using alias
- getting/setting task attachments
- getting/setting task comments

Form and Image Service
- getting process and task forms
- getting process and process instance images

Thanks,

Marian

sutaakar commented on kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java:
```diff
@@ -254,4 +277,183 @@ public void testStartProcessInDifferentDeploymentWithContainerId() throws Except
+        client.createContainer(CONTAINER_ID_101, containerResource);
+
+        try {
+            processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION_2);
```
Please use RestJmsSharedBaseIntegrationTest.assertClientException for negative tests to check response HTTP status and error messages.

MarianMacik replied:
Fixed now, thanks for notifying me about this feature!

sutaakar commented on kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java:
```diff
@@ -254,4 +275,178 @@ public void testStartProcessInDifferentDeploymentWithContainerId() throws Except
+        client.createContainer(CONTAINER_ID_101, containerResource);
+
+        assertClientException(
+                () -> processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION_2), 500, "No process available with given id");
```
If user tries to start not existing process then I would expect Kie server to return HTTP 400 or 404, definitely not 500. This looks like a bug.

tsurdilo replied:
can you raise a jira for this ? i can fix like we did for the previous tests that found issues with return codes.

mswiderski commented:
Better to not use BPMSPL jira in commit message that is not publicly available

MarianMacik commented:
@mswiderski sorry for the BPMSPL Jira, I was unable to find community one, but I have found it finally.

@tsurdilo Here is the [JIRA](https://issues.jboss.org/browse/JBPM-6023) for the little bug with HTTP code. I will fix my test to use 404 since it seems to be the most appropriate way.

rsynek commented on kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java:
```diff
@@ -254,4 +275,178 @@ public void testStartProcessInDifferentDeploymentWithContainerId() throws Except
+    public void testDifferentGAVsWithAlias() throws Exception {
+        Long oldPid = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION);
+        assertNotNull(oldPid);
+        assertTrue(oldPid > 0);
```
Not sure what information about the value will `assertTrue` give you in case it's not greater than 0. If you care about the value, there are assertions for numbers. 
And in general, using AssertJ simplifies assertions in tests. Have you considered using the library?

MarianMacik replied:
In fact, I considered it but already existing tests have been using these asserts so I didn't want to bring in an inconsistency. So should I rewrite it all to AssertJ?

MarianMacik replied:
OK, after discussion I will rewrite it with AssertJ.

rsynek commented on kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java:
```diff
@@ -254,4 +275,178 @@ public void testStartProcessInDifferentDeploymentWithContainerId() throws Except
+        List<String> availableSignals = processClient.getAvailableSignals(CONTAINER_ALIAS, processInstanceId);
+        assertNotNull(availableSignals);
+        assertEquals(2, availableSignals.size());
+        assertTrue(availableSignals.contains("Signal1"));
```
Another assertions where you can benefit from AssertJ - which offers nice assertion methods for collections.

MarianMacik commented:
I have rewritten asserts to AssertJ equivalents and fixed the test to expect 404. Therefore, there should be 3 tests failing (3 different marshalling formats) until @tsurdilo fixes the bug.

tsurdilo commented:
@MarianMacik thank for raising the jra - working on it

tsurdilo commented:
@MarianMacik prs for JBPM-6023:
https://github.com/kiegroup/droolsjbpm-integration/pull/995
https://github.com/kiegroup/jbpm/pull/863

-- could you please review? 
I added parts of your additions in BPMN2DataServicesTest but not all of your changes in this pr. once this is merged you will need to rebase and fix conflicts to add the rest of all your stuff from this pr. thanks!

MarianMacik commented:
I have pushed final version of the tests. Once they pass this can be merged.
